contracts/DollarToken.sol

```diff
@@ -86,10 +88,20 @@ contract DollarToken is Owned {
     {
         // TODO:
         //  receiveFee
-        //  allowWithdrawal
+        reserveFor(_beneficiaryAddr);
         Queue(queue).addBeneficiary(_beneficiaryAddr, msg.value);
     }
 
+    function reserveFor(address _participant)
+        payable
+    {
```

Comment:
@chatch 
> could be marked private to block outside calls

Tried that - this doesn't work because of Solidity limitations with CALL vs. DELEGATECALL, I believe? I tried marking it `private` and `internal` and neither compile, I think because internal calls can't be payable.

But I am fuzzy on the details of this